src/transforms/filter.jl

```diff
@@ -0,0 +1,44 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(f)
+
+The transform that filters the rows based on a given function.
+"""
+struct Filter{F} <: Transform
+  pred::F
+end
+
+isrevertible(::Type{Filter}) = true
+
+function apply(transform::Filter, input_table)
```

Comment:
Again, we don't use underscores in variable names. Please name the input table as `table` like in all other transforms.